File: src/lib/git.ts
/** One staged or unstaged working-tree entry returned by Tauri. */
export interface GitFileChange {
  path: string;
  oldPath: string | null;
  status: string;
  staged: boolean;
  untracked: boolean;
}

/** Repository identity and current working-tree changes. */
export interface GitChangesSnapshot {
  root: string;
  branch: string;
  files: GitFileChange[];
}

/** Stable identity for entries that may exist in both staged and unstaged groups. */
export function gitChangeKey(file: GitFileChange): string {
  return `${file.staged ? "staged" : "working"}:${file.path}`;
}

File: src-tauri/src/models/git.rs
//! Serializable Git working-tree models exposed to the frontend.

use serde::Serialize;

/// One staged or unstaged working-tree change.
#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GitFileChange {
    pub path: String,
    pub old_path: Option<String>,
    pub status: String,
    pub staged: bool,
    pub untracked: bool,
}

/// Current repository identity and changed files.
#[derive(Debug, Clone, PartialEq, Eq, Serialize)]
#[serde(rename_all = "camelCase")]
pub struct GitChangesSnapshot {
    pub root: String,
    pub branch: String,
    pub files: Vec<GitFileChange>,
}
